library/src/main/java/com/fstyle/library/DialogInit.java

```diff
@@ -0,0 +1,512 @@
+        WindowManager wm = dialog.getWindow().getWindowManager();
+        Display display = wm.getDefaultDisplay();
+        Point size = new Point();
+        display.getSize(size);
+        final int windowWidth = size.x;
+        final int windowHeight = size.y;
+
+        final int windowVerticalPadding = builder.context.getResources()
+                .getDimensionPixelSize(R.dimen.md_dialog_vertical_margin);
+        final int windowHorizontalPadding = builder.context.getResources()
+                .getDimensionPixelSize(R.dimen.md_dialog_horizontal_margin);
+        final int maxWidth =
+                builder.context.getResources().getDimensionPixelSize(R.dimen.md_dialog_max_width);
+        final int calculatedWidth = windowWidth - (windowHorizontalPadding * 2);
+
+        dialog.view.setMaxHeight(windowHeight - windowVerticalPadding * 2);
+        WindowManager.LayoutParams lp = new WindowManager.LayoutParams();
+        lp.copyFrom(dialog.getWindow().getAttributes());
+        lp.width = Math.min(maxWidth, calculatedWidth);
+        dialog.getWindow().setAttributes(lp);
+    }
+
+    private static void fixCanvasScalingWhenHardwareAccelerated(ProgressBar pb) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            // Canvas scaling when hardware accelerated results in artifacts on older API levels, so
+            // we need to use software rendering
+            if (pb.isHardwareAccelerated() && pb.getLayerType() != View.LAYER_TYPE_SOFTWARE) {
+                pb.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
+            }
+        }
+    }
+
+    private static void setupProgressDialog(final MaterialDialog dialog) {
+        final MaterialDialog.Builder builder = dialog.builder;
+        if (builder.indeterminateProgress || builder.progress > -2) {
+            dialog.progressBar = (ProgressBar) dialog.view.findViewById(android.R.id.progress);
+            if (dialog.progressBar == null) {
+                return;
+            }
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
+                if (builder.indeterminateProgress) {
+                    if (builder.indeterminateIsHorizontalProgress) {
+                        IndeterminateHorizontalProgressDrawable d =
+                                new IndeterminateHorizontalProgressDrawable(builder.getContext());
+                        d.setTint(builder.widgetColor);
+                        dialog.progressBar.setProgressDrawable(d);
+                        dialog.progressBar.setIndeterminateDrawable(d);
+                    } else {
+                        IndeterminateCircularProgressDrawable d =
+                                new IndeterminateCircularProgressDrawable(builder.getContext());
+                        d.setTint(builder.widgetColor);
+                        dialog.progressBar.setProgressDrawable(d);
+                                    dialog.progressBar.setIndeterminateDrawable(d);
```

Comment:
ERROR: 'else' child have incorrect indentation level 36, expected level should be 24.